src/runtime/internal/websocket.ts

```diff
@@ -79,9 +80,14 @@ export function useContentWebSocket() {
     }
 
     // WebSocket Base URL
-    const wsURL = new URL(`${(useRuntimeConfig().public.content as { wsUrl: string }).wsUrl}ws`)
+    const wsURL = new URL(joinURL((useRuntimeConfig().public.content as { wsUrl: string }).wsUrl, 'ws'))
     wsURL.protocol = window.location.protocol === 'https:' ? 'wss:' : 'ws:'
-    wsURL.hostname = window.location.hostname
+    if (wsURL.hostname === 'localhost') {
+      // If the hostname is localhost, use the hostname of the current window
+      // This is useful for local development with a reverse proxy
+      // As alternative, you can use the `publicURL` option to set the public URL of the WebSocket server
+      wsURL.hostname = window.location.hostname
+    }
```

Comment:
For me, this code should be removed. If we specifically want to force `localhost` via the config `content.watch.publicURL = 'https://localhost'`, it gets replaced by the current route. In this case, we should use `publicURL`.

Reply:
I don’t think browsers block connecting to `localhost` from another domain. As far as I know, WebSockets don’t have CORS validation in the same way as HTTP requests.

Ref: [StackOverflow – Why is there no same origin policy for WebSockets?](https://stackoverflow.com/questions/23674199/why-is-there-no-same-origin-policy-for-websockets-why-can-i-connect-to-ws-loc)

Reply:
Before this commit (https://github.com/nuxt/content/pull/3344/files
), with a proxy the hostname was always localhost by default, and for me it worked fine.

Reply:
Makes sense, thanks